validation/superchain-version_test.go

```diff
@@ -382,7 +389,18 @@ func checkMatch(s, c reflect.Value) bool {
 		}
 
 		if field.String() != c.Field(i).String() {
-			return false
+			if !isTestnet {
+				return false
+			}
+
+			// testnets are permitted to have contract versions that are newer than what's specified in the standard config
+			// testnets may NOT have contract versions that are older.
+			min := spr.CanonicalizeSemver(field.String())
+			current := spr.CanonicalizeSemver(c.Field(i).String())
+			if semver.Compare(min, current) > 0 {
+				return false
+			}
```

Comment:
This logic seems right when we are comparing semantic version strings, but this helper is also used when comparing bytecode hashes, where the semver comparison doesn't make sense. 

To fix that, we can either:
1. Store bytecode hashes for anticipated versions of contracts which maybe rolled out to testnets
2. Skip the bytecode hash check for testnets.

Reply:
I think (2) probably makes the most sense. We're not performing BHIC for testnets so I think we can skip the bytecode checking here too.